platform/openide.util.ui/src/org/openide/util/ImageUtilities.java

```diff
@@ -117,18 +125,27 @@
      * {@link Icon#paintIcon(Component, Graphics, int, int)} when converting an {@code Icon} to an
      * {@code Image}. See comment in {@link #icon2ToolTipImage(Icon, URL)}.
      */
-    private static volatile Component dummyIconComponent;
+    private static volatile Component dummyIconComponentLabel;
+
+    /**
+     * Second dummy component. Some {@link Icon#paintIcon(java.awt.Component, java.awt.Graphics, int, int)} are very picky and downcast the
+     * Component to a specific subclass. JCheckBox will satisfy checkboxes, abstract buttons etc. Will not eliminate all cases, but helps.
+     * 
+     */
+    private static volatile Component dummyIconComponentButton;
 
     static {
         /* Could have used Mutex.EVENT.writeAccess here, but it doesn't seem to be available during
         testing. */
         if (EventQueue.isDispatchThread()) {
-            dummyIconComponent = new JLabel();
+            dummyIconComponentLabel = new JLabel();
+            dummyIconComponentButton = new JButton();
```

Comment:
This still looks suspicious - `dummyIconComponentButton` is a different type depending on which thread hits this first?

Reply:
Done - fixed & review feedback squashed into the feature commits.

Reply:
Thanks!